processor_test.go

```diff
@@ -201,6 +201,57 @@ func TestServiceProcessor_ProcessClientRequest_Streaming(t *testing.T) {
 	close(inputChan)
 }
 
+func TestServiceProcessor_ProcessClientRequest_Streaming_Multiple(t *testing.T) {
```

Comment:
> this test sends multiple requests simulating a single client sending messages over the same connection

yes

> do we need to test multiple clients sending multiple requests?

The `ProcessClientStreamRequest` is mostly pure, except it uses the global `p.handlers` (which is not thread-safe, but that's another story...) and `p.Context.server.Suite()` so no, there isn't a huge need to check for multiple clients sending multiple requests, as it would be the same as calling multiple times the function.